scripts/tree.py

```diff
@@ -0,0 +1,444 @@
+Usage: python3 tree.py <sqlite-database> <term-curie> > <html-file>
+
+Creates an HTML page containing the tree structure of the term & its annotations.
+HTML is written to stdout.
+
+The sqlite-database must be created by RDFTab (https://github.com/ontodev/rdftab.rs)
+and include 'statements' and 'prefixes' tables.
+
+The term-curie must use a prefix from the 'prefixes' table.
+"""
+
+
+def main():
+    p = ArgumentParser("tree.py", description="create an HTML page to display an ontology term")
+    p.add_argument("db", help="SQLite database")
+    p.add_argument("term", help="CURIE of ontology term to display")
+    args = p.parse_args()
+
+    treename = os.path.splitext(os.path.basename(args.db))[0]
+
+    with sqlite3.connect(args.db) as conn:
+        conn.row_factory = dict_factory
+        cur = conn.cursor()
+        sys.stdout.write(terms2rdfa(cur, treename, [args.term]))
+
+
+def curie2href(curie):
+    """Convert a CURIE to an HREF"""
+    # TODO - this will replace curie2iri
+    return f"?term={curie}".replace("#", "%23")
+
+
+def curie2iri(prefixes, curie):
+    """Convert a CURIE to IRI"""
+    for prefix, base in prefixes:
+        if curie.startswith(prefix + ":"):
+            return curie.replace(prefix + ":", base)
+    raise Exception(f"No matching prefix for {curie}")
+
+
+def dict_factory(cursor, row):
+    """Create a dict factory for sqlite cursor"""
+    d = {}
+    for idx, col in enumerate(cursor.description):
+        d[col[0]] = row[idx]
+    return d
+
+
+def term2tree(data, treename, term_id):
+    """Create a hiccup-style HTML hierarchy vector for the given term."""
+    if treename not in data or term_id not in data[treename]:
+        return ""
+
+    tree = data[treename][term_id]
+    child_labels = []
+    for child in tree["children"]:
+        child_labels.append([child, data["labels"].get(child, child)])
+    child_labels.sort(key=lambda x: x[1].lower())
+
+    max_children = 100
+    children = []
+    for child, label in child_labels:
+        if child not in data[treename]:
+            continue
+        predicate = "rdfs:subClassOf"
+        oc = child
+        object_label = tree_label(data, treename, oc)
+        o = ["a", {"rev": predicate, "resource": oc}, object_label]
+        attrs = {}
+        if len(children) > max_children:
+            attrs["style"] = "display: none"
+        children.append(["li", attrs, o])
+        if len(children) == max_children:
+            total = len(tree["children"])
+            attrs = {"href": "javascript:show_children()"}
+            children.append(
+                ["li", {"id": "more"}, ["a", attrs, f"Click to show all {total} ..."]]
+            )
+    children = ["ul", {"id": "children"}] + children
+    if len(children) == 0:
+        children = ""
+
+    hierarchy = ["ul", ["li", tree_label(data, treename, term_id), children]]
+    i = 0
+    parents = tree["parents"]
+    if parents:
+        node = parents[0]
+        if term_id != "NCBITaxon:1":
```

Comment:
We need to generalize this root. Suggestions?

Reply:
I don't think we need this `if term_id != ...` statement; I removed it and it doesn't look like there's any unintended behavior, but what do you think?

Reply:
I'm not sure. It's a recursive loop that hung on me a few times. Get rid of it and we'll see! :^)